src/testutils.rs

```diff
@@ -137,6 +148,17 @@ impl DatabaseConnection for DummyDbConn {
         }
     }
 
+    fn spent_coins<'a>(&mut self, outpoints: &[(bitcoin::OutPoint, bitcoin::Txid, u32)]) {
+        for (op, spend_txid, time) in outpoints {
+            let mut db = self.db.write().unwrap();
+            let spent = &mut db.coins.get_mut(op).unwrap();
+            assert!(spent.spend_txid.is_none());
+            assert!(spent.spent_at.is_none());
```

Comment:
`spend_txid` shouldn't be `None` at the time you are confirming the spend transaction, it looks like a copy paste of the above method.